File: android/app/src/main/java/com/alandaitch/anden/data/store/AppSettings.kt
package com.alandaitch.anden.data.store

import android.content.Context
import android.content.SharedPreferences
import com.alandaitch.anden.AndenApp

// Ajustes de la app. Persiste en SharedPreferences.
class AppSettings(context: Context = AndenApp.appContext) {

    private val prefs: SharedPreferences =
        context.getSharedPreferences("anden.settings", Context.MODE_PRIVATE)

    var notifDemorasEnabled: Boolean
        get() = prefs.getBoolean("settings.notifDemoras", false)
        set(value) { prefs.edit().putBoolean("settings.notifDemoras", value).apply() }

    var seguirServicioId: String?
        get() = prefs.getString("settings.seguirServicioId", null)
        set(value) { prefs.edit().putString("settings.seguirServicioId", value).apply() }

    var onboardingDone: Boolean
        get() = prefs.getBoolean("settings.onboardingDone", false)
        set(value) { prefs.edit().putBoolean("settings.onboardingDone", value).apply() }

    companion object {
        val shared: AppSettings by lazy { AppSettings() }
    }
}

File: Shared/Store/AppSettings.swift
import Foundation
import Observation

// Ajustes de la app. Persiste en App Group.
@Observable
final class AppSettings {
    static let shared = AppSettings()

    private let defaults: UserDefaults

    var notifDemorasEnabled: Bool {
        didSet { defaults.set(notifDemorasEnabled, forKey: "settings.notifDemoras") }
    }
    var seguirServicioId: String? {
        didSet { defaults.set(seguirServicioId, forKey: "settings.seguirServicioId") }
    }
    var onboardingDone: Bool {
        didSet { defaults.set(onboardingDone, forKey: "settings.onboardingDone") }
    }

    init(defaults: UserDefaults? = nil) {
        let d = defaults ?? (UserDefaults(suiteName: "group.com.alandaitch.anden") ?? .standard)
        self.defaults = d
        self.notifDemorasEnabled = d.bool(forKey: "settings.notifDemoras")
        self.seguirServicioId = d.string(forKey: "settings.seguirServicioId")
        self.onboardingDone = d.bool(forKey: "settings.onboardingDone")
    }
}
